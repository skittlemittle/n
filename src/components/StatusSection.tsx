import * as React from "react";
import styled from "styled-components";

// its just a string shut up about "invalid syntax"
const Triangle = styled.div`
  width: 0;
  height: 0;
  margin: 0;
  border-top: 10px solid transparent;
  border-${(props) => props.dir}: 10px solid
    ${(props) => props.color};
  border-bottom: 10px solid transparent;
`;

const Rectangle = styled.div`
  font-family: "Inter";
  font-style: normal;
  font-weight: 600;
  display: flex;
  align-items: center;
  padding: 0 0.5em;
  height: 20px;
  min-width: 50px;
  margin: 0;
`;

const RectRight = styled(Rectangle)`
  margin-left: -10px;
  padding-left: 1em;
`;
const RectLeft = styled(Rectangle)`
  margin-right: -10px;
  padding-right: 1em;
`;

interface props {
  children: React.ReactNode;
  dir: "left" | "right";
  textColor: string;
  color: string;
  index: number;
}

/** container to hold stuff on the status line */
const StatusSection = (props: props) => {
  if (props.dir === "left") {
    return (
      <div style={{ display: "flex", zIndex: 100 - props.index }}>
        <Triangle dir={"right"} color={props.color} />
        <RectLeft
          style={{ backgroundColor: props.color, color: props.textColor }}
        >
          {props.children}
        </RectLeft>
      </div>
    );
  } else {
    return (
      <div style={{ display: "flex", zIndex: 100 - props.index }}>
        <RectRight
          style={{ backgroundColor: props.color, color: props.textColor }}
        >
          {props.children}
        </RectRight>
        <Triangle dir={"left"} color={props.color} />
      </div>
    );
  }
};

export default StatusSection;
