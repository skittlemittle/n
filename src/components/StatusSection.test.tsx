import React from "react";
import renderer from "react-test-renderer";
import StatusSection from "./StatusSection";

it("statusline section renders", () => {
  const tree1 = renderer
    .create(
      <StatusSection
        dir={"left"}
        textColor={"#fbf1c7"}
        color={"#7c6f64"}
        index={1}
      >
        V-LINE
      </StatusSection>
    )
    .toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer
    .create(
      <StatusSection
        dir={"right"}
        textColor={"#9d0006"}
        color={"#79740e"}
        index={4}
      >
        V-LINE
      </StatusSection>
    )
    .toJSON();
  expect(tree2).toMatchSnapshot();
});
