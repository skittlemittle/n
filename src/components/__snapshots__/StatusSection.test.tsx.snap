// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statusline section renders 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "zIndex": 99,
    }
  }
>
  <div
    className="sc-bdvvtL jMyhKK"
    color="#7c6f64"
    dir="right"
  />
  <div
    className="sc-gsDKAQ sc-hKwDye gxEtIK krXsrv"
    style={
      Object {
        "backgroundColor": "#7c6f64",
        "color": "#fbf1c7",
      }
    }
  >
    V-LINE
  </div>
</div>
`;

exports[`statusline section renders 2`] = `
<div
  style={
    Object {
      "display": "flex",
      "zIndex": 96,
    }
  }
>
  <div
    className="sc-gsDKAQ sc-dkPtRN gxEtIK XIEMA"
    style={
      Object {
        "backgroundColor": "#79740e",
        "color": "#9d0006",
      }
    }
  >
    V-LINE
  </div>
  <div
    className="sc-bdvvtL dvcpCb"
    color="#79740e"
    dir="left"
  />
</div>
`;
